rotkehlchen/externalapis/beaconchain/service.py

```diff
@@ -360,3 +373,81 @@ def get_eth1_address_validators(self, address: ChecksumEvmAddress) -> list[Valid
             ) from e
         return validators
+
+    def query_validator_daily_stats(
+            self,
+            validator_index: int,
+            last_known_timestamp: Timestamp,
+            exit_ts: Timestamp | None,
+    ) -> list[ValidatorDailyStats]:
+        """
+        May raise:
+        - RemoteError if we can't query beaconcha.in or if the data is not in the expected format
+        """
+        if exit_ts is not None and last_known_timestamp > exit_ts:
+            return []  # nothing new to add
+
+        data = cast(list[dict[str, Any]], self._query(
+            module='validator',
+            endpoint='stats',
+            encoded_args=str(validator_index),
+        ))
+
+        timestamp = Timestamp(0)
+        stats: list[ValidatorDailyStats] = []
+        for day_data in data:
+            try:
+                start_effective_balance = day_data['start_effective_balance']
+                if start_effective_balance is None:
+                    start_effective_balance = ZERO
+                elif deserialize_int(start_effective_balance) == ZERO:
+                    continue  # validator exited
+
+                end_effective_balance = None
+                if day_data['end_effective_balance'] is not None:
+                    end_effective_balance = deserialize_int(day_data['end_effective_balance'])
+                withdrawals_amount = deserialize_int(day_data['withdrawals_amount'])
+
+                try:
+                    timestamp = create_timestamp(
+                        datestr=f"{day_data['day_start']}",
+                        formatstr='%Y-%m-%dT%I:%M:%S%z',
+                    )
+                except ValueError as e:
+                    raise RemoteError(
+                        f'Failed to parse {day_data["day_start"]} to timestamp',
+                    ) from e
+
+                if timestamp <= last_known_timestamp:
+                    return stats  # we are done
+
+                end_balance = start_balance = 0
+                if (raw_end_balance := day_data['end_balance']) is not None:
+                    end_balance = deserialize_int(raw_end_balance)
+
+                if (raw_start_balance := day_data['start_balance']) is not None:
+                    start_balance = deserialize_int(raw_start_balance)
+
+            except KeyError as e:
+                log.error(f'Missing key {e} in daily stats for {validator_index=}. Skipping day')
+                continue
+
+            except DeserializationError as e:
+                log.error(f'Failed to deserialize daily stats for {validator_index=} due to {e}. Skipping day')  # noqa: E501
+                continue
+
+            # end balance in the api is the actual balance of the validator at the end of the day,
+            # start balance is the actual balance of the validator at the start of the day,
+            # end_effective_balance and start_effective_balance are the effective balance
```

Comment:
Here I meant to also explain what the heck effective balance is, and why use it. I still don't get it. No need to explain that end_XXX, means XXX at the end of the day. That's obvious.